dj/sql/parsing/ast.py

```diff
@@ -32,9 +35,21 @@ def flatten(maybe_iterables: Any) -> Iterator:
     )
 
 
-class Node(ABC):
+class DJEnum(Enum):
     """
-    Base class for all DJ AST nodes.
+    A DJ AST enum
+    """
+
+    def __repr__(self) -> str:
+        return str(self)
+
+# typevar used for node methods that return self
+# so the typesystem can correlate the self type with the return type
+TNode = TypeVar("TNode", bound="Node")
```

Comment:
nit: s/TNode/TypeNode (I think it'd be easier to remember what it is when seen in other places)